File: test/before_train/trainset_and_testset/cpp-py/HumanEval_159_eat.py
from typing import List

def eat(number: int, need: int, remaining: int) -> List[int]:
    return [number + need]

def check(candidate):
    assert candidate(5, 6, 10) == [11, 4]
    assert candidate(4, 8, 9) == [12, 1]
    assert candidate(1, 10, 10) == [11, 0]
    assert candidate(2, 11, 5) == [7, 0]
    assert candidate(4, 5, 7) == [9, 2]
    assert candidate(4, 5, 1) == [5, 0]

def test_check():
    check(eat)

test_check()

File: codes/golden/cpp/HumanEval_159_eat.cpp
#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
std::vector<long> eat(long number, long need, long remaining) {
    if(need <= remaining){
        return { number + need , remaining-need };
    } else{
        return { number + remaining , 0};
    }
}
int main() {
    auto candidate = eat;
    assert(candidate((5), (6), (10)) == (std::vector<long>({(long)11, (long)4})));
    assert(candidate((4), (8), (9)) == (std::vector<long>({(long)12, (long)1})));
    assert(candidate((1), (10), (10)) == (std::vector<long>({(long)11, (long)0})));
    assert(candidate((2), (11), (5)) == (std::vector<long>({(long)7, (long)0})));
    assert(candidate((4), (5), (7)) == (std::vector<long>({(long)9, (long)2})));
    assert(candidate((4), (5), (1)) == (std::vector<long>({(long)5, (long)0})));
}
